LinkRelay/plugin.py

```diff
@@ -302,11 +303,19 @@ def send(s):
                 self.log.info('LinkRelay:  I\'m not in in %s on %s' %
                               (relay.targetChannel, relay.targetNetwork))
             else:
-                if isPrivmsg or \
+                if isPrivmsg and 'draft/relaymsg' in targetIRC.state.capabilities_ls \
+                    and targetIRC.state.channels[relay.targetChannel].isOp(targetIRC.nick) \
+                    and self.registryValue('useRelayMsg', relay.targetChannel):
+                    seperator = targetIRC.state.capabilities_ls['draft/relaymsg']
+                    f = functools.partial(ircmsgs.relaymsg, relay.targetChannel, 
+                    '{}{}{}'.format(args['nick'], seperator, irc.network))
+                    args['draft/relaymsg'] = True
+                    s = args['text']
+                elif isPrivmsg or \
                         self.registryValue('nonPrivmsgs', channel) == 'privmsg':
-                    f = ircmsgs.privmsg
+                    f = functools.partial(ircmsgs.privmsg, relay.targetChannel)
                 elif self.registryValue('nonPrivmsgs', channel) == 'notice':
-                    f = ircmsgs.notice
+                    f = functools.partial(ircmsgs.notice, relay.targetChannel)
```

Comment:
Why this change?

Reply:
This is because the args of `ircmsgs.privmsg` and `ircmsgs.notice` are `recipient` and `s` while that of `relaymsg` are `recipient`, `spoofed_nick` and `s`.

I could reorder the args of the relaymsg function to be `(spoofed_nick, recipient, s)` and then only apply `partial` when the `msg` to be sent is a relaymsg (`f = functools.partial(ircmsgs.relaymsg, sppofed_nick)` and `f = ircmsgs.notice` for others etc) and use the old `msg = f(relay.targetChannel, s)`. But then the order of args would not be same as in the spec (not needed, but it seems cleaner to follow that order)

So it seemed cleaner to me to apply partial all functions to sort out the order and only call `msg = f(s)`.

Reply:
hmm ok then